core/src/main/java/de/danoeh/antennapod/core/service/download/NewEpisodesNotification.java

```diff
@@ -0,0 +1,88 @@
+package de.danoeh.antennapod.core.service.download;
+
+import android.app.PendingIntent;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.res.Resources;
+
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import de.danoeh.antennapod.core.R;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedItem;
+import de.danoeh.antennapod.core.feed.FeedPreferences;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.danoeh.antennapod.core.util.gui.NotificationUtils;
+
+public class NewEpisodesNotification {
+    private final List<Feed> feeds = new ArrayList<>();
+    private final Map<Long, FeedItem> lastItemsMap = new HashMap<>();
```

Comment:
How about storing just the FeedItem id here instead of the whole FeedItem? That avoids storing data unnecessarily.